src/main/java/dev/hephaestus/atmosfera/client/sound/modifiers/implementations/PercentBiomeModifier.java

```diff
@@ -97,20 +101,23 @@ private record Factory(float min, float max, ImmutableCollection<Identifier> bio
 
         @Override
         public AtmosphericSoundModifier create(World world) {
-            ImmutableCollection.Builder<Biome> biomes = ImmutableList.builder();
+            ImmutableCollection.Builder<Pair<Biome, RegistryEntry<Biome>>> biomes = ImmutableList.builder();
+
+            Registry<Biome> biomeRegistry = world.getRegistryManager().get(Registry.BIOME_KEY);
 
             for (Identifier id : this.biomes) {
-                Biome biome = world.getRegistryManager().get(Registry.BIOME_KEY).get(id);
+                Biome biome = biomeRegistry.get(id);
 
                 if (biome != null) {
-                    biomes.add(biome);
+                    RegistryEntry<Biome> biomeEntry = biomeRegistry.entryOf(biomeRegistry.getKey(biome).get());// should never throw
+                    biomes.add(Pair.of(biome, biomeEntry));
```

Comment:
Build a list of `Pair<Biome, RegistryEntry<Biome>>`, since we need the `RegistryEntry` to check the biome's tags.